src/objects/Mesh.js

```diff
@@ -245,24 +245,67 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
 				var index = geometry.index;
 				var position = geometry.attributes.position;
 				var uv = geometry.attributes.uv;
-				var i, l;
+				var groups = geometry.groups;
+				var drawRange = geometry.drawRange;
+				var drawStart = drawRange.start;
+				var drawEnd = drawStart + drawRange.count;
+				var i, j, il, jl;
+				var group, groupMaterial;
 
 				if ( index !== null ) {
 
 					// indexed buffer geometry
 
-					for ( i = 0, l = index.count; i < l; i += 3 ) {
+					if ( Array.isArray( material ) ) {
 
-						a = index.getX( i );
-						b = index.getX( i + 1 );
-						c = index.getX( i + 2 );
+						for ( i = 0, il = groups.length; i < il; i ++ ) {
 
-						intersection = checkBufferGeometryIntersection( this, raycaster, ray, position, uv, a, b, c );
+							group = groups[ i ];
+							groupMaterial = material[ group.materialIndex ];
 
-						if ( intersection ) {
+							for ( j = group.start, jl = ( group.start + group.count ); j < jl; j += 3 ) {
 
-							intersection.faceIndex = Math.floor( i / 3 ); // triangle number in indices buffer semantics
-							intersects.push( intersection );
+								if ( j >= drawStart && ( j + 2 ) < drawEnd ) {
```

Comment:
I've looked into this but the situation in `.renderBufferDirect()` is different than in `.raycast()`. `.renderBufferDirect()` already gets the correct group and does not have to iterate over all existing groups like in `.raycast()`. This makes a direct comparison of both methods complicated. Besides, i'm not sure if we need to regard `rangeFactor` in `.raycast()`.